src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2135,11 +2137,17 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		if (!is_element_type_primitive (fsig->params [0]) || !is_element_type_primitive (fsig->params [1]))
 			return NULL;
 
+#if defined(TARGET_S390X)
+		if (!mono_hwcap_s390x_has_ve1 && ((id == SN_Max) || (id == SN_Min) || (id == SN_MaxNative) || (id == SN_MinNative) || (id !=SN_Xor) || (id != SN_BitwiseAnd) || (id != SN_BitwiseOr)) && arg0_type == MONO_TYPE_R4)
```

Comment:
This logic is incorrect.  `(id != SN_BitwiseAnd) || (id != SN_BitwiseOr)` is always true.

In any case, would it not be simpler to move the check to before the whole switch statement in the first place?  Without `ve1` we have *no* vector float support at all, so can't we simply reject any operation on `MONO_TYPE_R4` without checking which one it is?   Same comment would apply throughout this file.

(In theory, we could implement the bitwise ops.  But in the big picture, it's not worthwhile special-casing those.)